frontend/src/components/Tree/Tree.jsx

```diff
@@ -0,0 +1,139 @@
+import React, { useState, useRef, useEffect } from "react";
+import { select, hierarchy, tree, linkHorizontal } from "d3";
+import ResizeObserver from "resize-observer-polyfill";
+import "./Tree.module.css";
+
+const useResizeObserver = (ref) => {
+  const [dimensions, setDimensions] = useState(null);
+  useEffect(() => {
+    const observeTarget = ref.current;
+    const resizeObserver = new ResizeObserver((entries) => {
+      entries.forEach((entry) => {
```

Comment:
Do we always expect there to be one entry here or can there be several in some cases?

Reply:
I'm not sure actually since Sultan wrote this part. @sultansidhu

Reply:
hmm its been quite long since I wrote that 😰 would have to revisit it. So if it isnt an urgent thing can perhaps merge this one for now and get back to it at a later stage?

Reply:
ok sounds good (y) I think it should be ok but best to check at some point